mitiq/benchmarks/tests/test_qpe_circuits.py

```diff
@@ -0,0 +1,204 @@
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[2], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[1], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[1], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                cirq.SWAP(qubits[0], qubits[2]),
+            ),
+            cirq.Moment(
+                cirq.H(qubits[0]),
+                cirq.H(qubits[1]),
+                cirq.H(qubits[2]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**-0.25).on(qubits[0], qubits[1]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**-0.25).on(qubits[0], qubits[2]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**-0.25).on(qubits[1], qubits[2]),
+            ),
+        ]
+    )
+    assert _equal(generated_circuit, expected_circuit)
+
+
+def test_phase_angle_estimation():
+    def _binary_to_int(input_array):
+        result_string = ""
+        for i in range(len(input_array)):
+            result_string += str(input_array[i])
+            result = int(result_string, 2)
+        return result
+
+    n_eigstate_reg = 3
+    # initialize the eigenstate reg to |+> by using H after Pauli X Gate
+    # + add measurements on ereg register
+    # expect eigenvalue to be +1
+    qubits = cirq.LineQubit.range(n_eigstate_reg + 1)
+    PauliX_on_0_qft_circuit = (
+        cirq.Circuit(cirq.H(qubits[3]))
+        + generate_qpe_circuit(3, cirq.X)
+        + cirq.measure(qubits[0:3])
+    )
+    simulator = cirq.Simulator()
+    result1 = simulator.run(PauliX_on_0_qft_circuit)
+    result_array1 = result1.records["q(0),q(1),q(2)"]
+    result1 = _binary_to_int(result_array1[0][0])
+    calculated_angle = result1 / 2**n_eigstate_reg
+    assert calculated_angle == 0
+
+    # initialize the eigenstate reg to |-> by using X and H after Pauli X Gate
+    # + add measurements on ereg register
+    # expect eigenvalue to be -1 (ignoring complex i)
+    qubits = cirq.LineQubit.range(n_eigstate_reg + 1)
+    PauliX_on_1_qft_circuit = (
+        cirq.Circuit(cirq.X(qubits[3]), cirq.H(qubits[3]))
+        + generate_qpe_circuit(3, cirq.X)
+        + cirq.measure(qubits[0:3])
+    )
+    simulator = cirq.Simulator()
+    result2 = simulator.run(PauliX_on_1_qft_circuit)
+    result_array2 = result2.records["q(0),q(1),q(2)"]
+    result2 = _binary_to_int(result_array2[0][0])
+    calculated_angle = result2 / 2**n_eigstate_reg
+    assert calculated_angle == 0.75
```

Comment:
Is this consistent with the expected eigenvalue which is `-1`  (according to your comment) ?   
Is the eigenvalue given by something like `exp(2 pi i calculated_angle)` ? 
If so, shouldn't `calculated_angle` be 0.5 ?

Reply:
The eignevalue with this test is `-i`. I was ignoring the `i` 

Maybe I should double check the calculations and the defined function.